tools/ips.py

```diff
@@ -29,32 +43,89 @@ def generate_dev(device_name, ip_addr):
     return os.system(cmd)
 
 
+def remove_dev(device_name, ip_addr):
+    return os.system('ip link delete ' + device_name)
+
+
+def add_addr_to_existing_dev(device_name, ip_addr):
+    if sys.platform.startswith('linux'):
+        cmd = 'ip addr add ' + ip_addr + ' dev ' + device_name + ' && '
+        cmd += 'ip link set ' + device_name + ' up'
+    else:
+        cmd = 'ifconfig ' + device_name + ' alias ' + ip_addr + '/32'
+    return os.system(cmd)
+
+
+def delete_addr_from_existing_dev(device_name, ip_addr):
+    if sys.platform.startswith('linux'):
+        cmd = 'ip addr del ' + ip_addr + '/32 dev ' + device_name
+    else:
+        cmd = 'ifconfig ' + device_name + ' -alias ' + ip_addr
+    return os.system(cmd)
+
+
 parser = argparse.ArgumentParser(description='Setting interfaces and generating Rust file with IPs.')
 parser.add_argument('--subnet', nargs='?', default='1.1.1.0/28', help='Subnet to generate', type=str)
-parser.add_argument('--file', nargs='?', default='ips.rs', help='Output file with IPs', type=str)
-parser.add_argument('--dev_prefix', nargs='?', default='test_eth', help='How to prefix dummy devs', type=str)
+parser.add_argument('--file', nargs='?', default='ips.rs', help='Output file with IPs (default ips.rs)', type=str)
+parser.add_argument('--op', nargs='?', default='add', help='add/remove operation', type=str)
+parser.add_argument('--dev_prefix', nargs='?', help='How to prefix dummy devs (Linux only)(def: test_dummyX)', type=str)
+parser.add_argument('--dev', nargs='?', help='Device to add addresses (Linux and Mac)(default: lo0)', type=str)
 
 args = parser.parse_args()
 
 subnet = args.subnet
 file = args.file
-dev = args.dev_prefix
+op = args.op
+
+if not sys.platform.startswith('linux') and not sys.platform.lower().startswith('darwin'):
+    print('Script work only under Linux or MacOS')
+    sys.exit(0)
+
+if args.dev_prefix is None and args.dev is None:
+    print('Either --dev_prefix or --dev must be specified')
+    sys.exit(0)
+
+if op not in ['add', 'remove']:
+    print('Operation must be either add or remove')
+    sys.exit(0)
+
+dev = None
+dev_prefix = None
+
+if op == 'add':
+    operation_dev = 'generate_dev'
+    operation_addr = 'add_addr_to_existing_dev'
+else:
+    operation_dev = 'remove_dev'
+    operation_addr = 'delete_addr_from_existing_dev'
+
+if args.dev is not None:
+    dev = args.dev
+else:
+    if args.dev_prefix is None:
+        dev = 'lo0'
```

Comment:
I don't know python, but wouldn't would this work here?
```suggestion
else if args.dev_prefix is None:
        dev = 'lo0'
```

Reply:
But now that I see this code above:
```py
if args.dev_prefix is None and args.dev is None:
    print('Either --dev_prefix or --dev must be specified')
    sys.exit(0)
```
Seems like we should never be able to enter this case.
I think we can simply use this instead:
```py
if args.dev is not None:
    dev = args.dev
```

Reply:
Removed.